### DISCUSSION TITLE
Formatting issues exporting to ODT

### DISCUSSION THREAD

**Discussion by wcjessen** [Zettlr/Zettlr#2623] [Q&A] [answered]
There are plenty of discussions for exporting MD documents to PDF, but it has proven difficult to find anything for ODT. There seems to be no difference between ODT and DocX, so I only mention my preferred ODT.

First, when I export to ODT, my settings for line spacing do not carry over. It works in my exported PDFs, but I need to export as an actual document (preferably ODT). Also, how do I set up 1.5 spacing? I have \doublespacing activated in the YAML Frontmatter header-includes, but I am not sure what wordage to use for 1.5. 

Second, paragraph indents sort of work in PDF (just not the first paragraph after headings), but not in ODT.

Finally, \newpage breaks work in PDF, but not in ODT.

Any help is greatly appreciated!

**Comment by nathanlesage** [comment, marked as answer]
You are right in that PDF is the most common export format so a lot of the info is revolving around that one. You are also right in that Docx and ODT exports don't fundamentally differ. Now, to the stuff you want to achieve.

Stuff like `\doublespacing` or `\newpage` are LaTeX commands, that is: they will only be recognized by the LaTeX engine, which doesn't participate in exports to other formats.

Going further, there are two ways in which you can define how certain exports work: templates and reference docs. For many export formats you need to (or rather, you may) define templates. For docx and odt exports, however, Pandoc requires a so-called "reference document". How you create one is described in the Pandoc manual. Please search for `--reference-doc=` on the site, then you will find the descriptions.

The default reference docs which Pandocs ships with can be found here: https://github.com/jgm/pandoc/tree/master/data/odt

Unfortunately, they are unzipped, so to get them, you'd have to download the folder, zip them, rename them to `.odt` or `.docx` and then open them. It sounds weird, and it is weird, but that's how it works. It might be worthy to offer a menu item in Zettlr to allow to do that easily in the future …

**Reply by wcjessen** [reply]
Thank you so much! I am not a techy at all, so some of the language still eludes me. Using this reference doc definitely changed how things looked, so it should be a matter of tweaking the formatting. Thanks again!

**Reply by nathanlesage** [reply]
Always a pleasure :)

**Reply by wcjessen** [reply]
I didn't want to spam the Q&A forum, so I'll try asking here. Even on the Pandoc website, there is very little information on what exactly you should do within that reference doc. I played around with [this guy's pre-made doc](https://www.autodidacts.io/convert-markdown-to-standard-manuscript-format-odts-docs-and-pdfs-with-pandoc/), and it DID modify my document, but it was not what I wanted at all. Do I have to essentially create a "fake" document, formatted exactly how I want, within that file? Thanks again!

**Reply by nathanlesage** [reply]
> Even on the Pandoc website, there is very little information on what exactly you should do within that reference doc.

Yes, that's unfortunate, but what you essentially need to do is adapt the *styles*, which will be applied to the elements. So for ODT you would need to adapt the formats (Paragraph, Heading, Body Text, Footnote Text, etc.) as well as for Word. So without editing the document itself, you ONLY need to adapt the formats. Here a screenshot from Word:

![image](https://user-images.githubusercontent.com/17251683/137188299-d56663a6-62d5-4bf1-a2a8-d595c7d7c979.png)

**Reply by wcjessen** [reply]
OK that makes sense. These were the last hurdles I had converting entirely to MD, so you rock!
